### PR TITLE
feat: P002271-5772 - Assign help items to products in UI

### PR DESCRIPTION
**Pull request by anetastepien**

### CODE REVIEW

File: src/app/help/help-detail/help-detail.component.ts [hunk 1]
```diff
@@ -41,14 +42,18 @@ export class HelpDetailComponent implements OnChanges {
    *  SAVING
    */
   public onSave() {
-    this.changeMode === 'NEW' ? this.createHelpItem() : this.updateHelpItem()
+    const helpObject = { ...this.helpFormComponent.formGroup.value }
+    helpObject.productName = helpObject.product.name
+    delete helpObject.product
+
+    this.changeMode === 'NEW' ? this.createHelpItem(helpObject) : this.updateHelpItem(helpObject)
```
Comment (markuczy) [inline src/app/help/help-detail/help-detail.component.ts:49]: Why is that needed?

File: src/app/help/help-detail/help-detail.component.ts [hunk 2]
```diff
@@ -41,14 +42,18 @@ export class HelpDetailComponent implements OnChanges {
...
   }
 
-  private createHelpItem() {
+  private createHelpItem(helpObject: any) {
```
Comment (markuczy) [inline src/app/help/help-detail/help-detail.component.ts:52]: Do not use any if possible

File: src/app/help/help-detail/help-detail.component.ts [hunk 3]
```diff
@@ -70,12 +75,12 @@ export class HelpDetailComponent implements OnChanges {
     }
   }
 
-  private updateHelpItem(): void {
-    if (this.helpFormComponent.formGroup.valid && this.appId && this.itemId) {
+  private updateHelpItem(helpObject: any): void {
```
Comment (markuczy) [inline src/app/help/help-detail/help-detail.component.ts:78]: Do not use any if possible

File: src/app/remotes/help-item-editor/help-item-editor.component.ts [hunk 4]
```diff
@@ -48,7 +49,8 @@ import { HelpItemEditorDialogComponent } from './help-item-editor-dialog/help-it
   providers: [
     HelpsInternalAPIService,
     PortalMessageService,
-    providePortalDialogService(),
+    DialogService,
+    PortalDialogService,
```
Comment (markuczy) [inline src/app/remotes/help-item-editor/help-item-editor.component.ts:53]: Do not import PrimeNG dialogService related services and use providePortalDialogService()

File: src/app/remotes/help-item-editor/help-item-editor.component.ts [hunk 5]
```diff
@@ -90,20 +92,20 @@ export class OneCXHelpItemEditorComponent implements ocxRemoteComponent {
...
       this.appStateService.currentPage$.asObservable(),
       this.appStateService.currentMfe$.asObservable()
     ]).pipe(
       map(([page, mfe]) => {
         if (page?.applicationId) return page.applicationId
-        if (mfe.appId) return mfe.appId
+        if (mfe.productName) return mfe.productName
         return ''
       })
     )
```
Comment (markuczy) [inline src/app/remotes/help-item-editor/help-item-editor.component.ts:105]: currentPage$.applicationId is no longer usable so I would only use mfe.productName

Comment (markuczy) [reply src/app/remotes/help-item-editor/help-item-editor.component.ts:105]: Please remove related tests too

File: src/app/remotes/show-help/show-help.component.ts [hunk 6]
```diff
@@ -87,20 +87,20 @@ export class OneCXShowHelpComponent implements ocxRemoteComponent {
         return router.routerState.snapshot.url.split('#')[0]
       })
     )
-    this.applicationId$ = combineLatest([
+    this.productName$ = combineLatest([
       this.appStateService.currentPage$.asObservable(),
       this.appStateService.currentMfe$.asObservable()
     ]).pipe(
       map(([page, mfe]) => {
         if (page?.applicationId) return page.applicationId
```
Comment (markuczy) [inline src/app/remotes/show-help/show-help.component.ts:95]: Same as for help-item-editor

**Review by markuczy — CHANGES_REQUESTED** [review]

**Review by markuczy — COMMENTED** [review]
Please ensure that all components are working as they did with app_id including remote components (top right corner of the page)
image

**Review by markuczy — COMMENTED** [review]
image
None product option is not highlighted in any way

File: src/app/remotes/help-item-editor/help-item-editor.component.ts (hunk 4 above)
Comment (anetastepien) [reply src/app/remotes/help-item-editor/help-item-editor.component.ts:53]: Merge error. Fixed

File: src/app/help/help-search/help-search.component.ts [hunk 7]
```diff
@@ -79,6 +99,15 @@ export class HelpSearchComponent implements OnInit {
     })
   }
 
+  private processProducts(productsPageResult: ProductsPageResult) {
+    this.products = productsPageResult.stream ?? []
+    if (this.products?.length === 0) {
+      this.msgService.info({ summaryKey: 'HELP_SEARCH.NO_PRODUCTS_AVAILABLE' })
+    }
+    this.products = this.products?.filter((product) => product !== null)
```
[the remark this answers is not in the document]
Comment (anetastepien) [reply src/app/help/help-search/help-search.component.ts:107]: resolved

File: src/app/help/help-detail/help-detail.component.ts (hunk 1 above)
Comment (anetastepien) [reply src/app/help/help-detail/help-detail.component.ts:49]: Till now we saved the form exactly as it was (a set of simple fields). After adding combobox, we bind it with an object containing name and displayName. However we still save only name, so before saving we map it

File: src/app/help/help-search/help-search.component.ts [hunk 8]
```diff
@@ -21,6 +29,9 @@ export class HelpSearchComponent implements OnInit {
   public actions: Action[] = []
   public helpItem: Help | undefined
   public results: Help[] = []
```
[the remark this answers is not in the document]
Comment (anetastepien) [reply src/app/help/help-search/help-search.component.ts:31]: removed

File: src/app/help/help-detail/help-detail.component.ts (hunk 2 above)
Comment (anetastepien) [reply src/app/help/help-detail/help-detail.component.ts:52]: fixed

File: src/app/help/help-search/help-search.component.ts [hunk 9]
```diff
@@ -21,6 +29,9 @@ export class HelpSearchComponent implements OnInit {
...
+  public resultsForDisplay: any[] = []
```
[the remark this answers is not in the document]
Comment (anetastepien) [reply src/app/help/help-search/help-search.component.ts:32]: fixed

File: src/app/help/help-detail/help-detail.component.ts (hunk 3 above)
Comment (anetastepien) [reply src/app/help/help-detail/help-detail.component.ts:78]: fixed

**Review by HenryT-CG — APPROVED** [review]
